src/dex/balancer-v2/balancer-v2.ts

```diff
@@ -328,50 +372,51 @@ export class BalancerV2EventPool extends StatefulEventSubscriber<PoolStateMap> {
   ): Promise<PoolStateMap> {
     const multiCallData = subgraphPoolBase
       .map(pool => {
-        let poolCallData = [
-          {
-            target: this.vaultAddress,
-            callData: this.vaultInterface.encodeFunctionData('getPoolTokens', [
-              pool.id,
-            ]),
-          },
-          {
-            target: pool.address,
-            callData: this.poolInterfaces['Weighted'].encodeFunctionData(
-              'getSwapFeePercentage',
-            ), // different function for element pool
-          },
-        ];
-
-        if (['MetaStable'].includes(pool.poolType)) {
-          poolCallData.push({
-            target: pool.address,
-            callData:
-              this.poolInterfaces['MetaStable'].encodeFunctionData(
-                'getScalingFactors',
-              ),
-          });
-        }
+        let poolCallData = [];
         if (
```

Comment:
Though the code is logically correct. 
It would be better to use `switch` instead of `if if if ..` because a pooltype will only call one `getOnChainCalls`. Same structure for decode. Another way would be just use an dict to store a map from poolType to poolImplementation in the constructor. Then simply do `dict[poolType].getOnChainCalls(...)/decodeOnChainCalls(...)`

Reply:
Refactored and also applied same logic to `getPricesPool`.